gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegistrationUnit.java

```diff
@@ -338,6 +351,124 @@ public void setSerDeType(String serDeType) {
     this.serDeType = Optional.of(serDeType);
   }
 
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((this.bucketColumns == null) ? 0 : this.bucketColumns.hashCode());
+    result = prime * result + ((this.columns == null) ? 0 : this.columns.hashCode());
+    result = prime * result + ((this.dbName == null) ? 0 : this.dbName.hashCode());
+    result = prime * result + ((this.inputFormat == null) ? 0 : this.inputFormat.hashCode());
+    result = prime * result + ((this.isCompressed == null) ? 0 : this.isCompressed.hashCode());
+    result = prime * result + ((this.isStoredAsSubDirs == null) ? 0 : this.isStoredAsSubDirs.hashCode());
+    result = prime * result + ((this.location == null) ? 0 : this.location.hashCode());
+    result = prime * result + ((this.numBuckets == null) ? 0 : this.numBuckets.hashCode());
+    result = prime * result + ((this.outputFormat == null) ? 0 : this.outputFormat.hashCode());
+    result = prime * result + ((this.serDeType == null) ? 0 : this.serDeType.hashCode());
+    result = prime * result + ((this.tableName == null) ? 0 : this.tableName.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
```

Comment:
These can be the default properties to compare. But for other cases, like hive distcp, properties like "location" will be customized.